skxray/core/roi.py

```diff
@@ -357,139 +355,87 @@ def roi_pixel_values(image, labels, index=None):
-        Each ROI is represented by a distinct label (i.e., integer).
-
+    mean_intensity : list
+        The mean intensity of each ROI for all `images`
+        Dimensions:
+            len(mean_intensity) == len(index)
+            len(mean_intensity[0]) == len(images)
     index : list
-        labels list
-        eg: 5 ROI's
-        index = [1, 2, 3, 4, 5]
-
-    Returns
-    -------
-    mean_intensity : array
-        mean intensity of each ROI for the set of images as an array
-        shape (len(images), number of labels)
-
+        The labels for each element of the `mean_intensity` list
     """
-    if labels.shape != images[0].shape[0:]:
-        raise ValueError("Shape of the images should be equal to"
-                         " shape of the label array")
+    if labeled_array.shape != images[0].shape[0:]:
+        raise ValueError(
+            "`images` shape (%s) needs to be equal to the labeled_array` shape"
+            "(%s)" % (images[0].shape, labeled_array.shape))
+    # handle various input for `index`
     if index is None:
-        index = np.arange(1, np.max(labels) + 1)
-
-    mean_intensity = np.zeros((images.shape[0], index.shape[0]))
+        index = list(np.unique(labeled_array))
+        index.remove(0)
+    try:
+        len(index)
+    except TypeError:
+        index = [index]
+    # not sure that this is needed
+    index = np.asarray(index)
+    # pre-allocate an array for performance
+    # might be able to use list comprehension to make this faster
+    mean_intensity = np.zeros((images.shape[0], len(index)))
     for n, img in enumerate(images):
-        mean_intensity[n] = ndim.mean(img, labels, index=index)
-
-    return mean_intensity, index
-
-
-def combine_mean_intensity(mean_int_list, index_list):
-    """
-    Combine mean intensities of the images(all images sets) for each ROI
-    if the labels list of all the images are same
-
-    Parameters
-    ----------
-    mean_int_list : list
-        mean intensity of each ROI as a list
-        shapes is: (len(images_sets), )
-
-    index_list : list
-        labels list for each image sets
-
-    img_set_names : list
-
-    Returns
-    -------
-    combine_mean_int : array
-        combine mean intensities of image sets for each ROI of labeled array
-        shape (number of images in all image sets, number of labels)
-
-    """
-    if np.all(map(lambda x: x == index_list[0], index_list)):
-        combine_mean_intensity = np.vstack(mean_int_list)
-    else:
-        raise ValueError("Labels list for the image sets are different")
-
-    return combine_mean_intensity
+        # use a mean that is mask-aware
+        mean_intensity[n] = ndim.mean(img, labeled_array, index=index)
+    # turn the 2-D array back into a list of arrays because the rows and
+    # columns have different units
+    data = [mean_intensity[:, i] for i in range(mean_intensity.shape[1])]
+    roi_labels = ['roi_%s' % idx for idx in index]
```

Comment:
What is the use for returning a list of strings here?  I think just returning `index` would be more useful here.